src/software/Localization/main_SfM_Localization.cpp

```diff
@@ -235,19 +255,27 @@ int main(int argc, char **argv)
   const int num_initial_intrinsics = sfm_data.GetIntrinsics().size(); 
 
   int total_num_images = 0;
-  for ( std::vector<std::string>::const_iterator iter_image = vec_image_new.begin();
-    iter_image != vec_image_new.end();
-    ++iter_image)
+  
+#ifdef OPENMVG_USE_OPENMP
+    const unsigned int nb_max_thread = omp_get_max_threads();
+    omp_set_num_threads(iNumThreads);
+    #pragma omp parallel for schedule(dynamic)
```

Comment:
What about something like
```
const unsigned int nb_max_thread = (iNumThreads == 0) ? 0 : omp_get_max_threads();
omp_set_num_threads(nb_max_thread);
#pragma omp parallel for schedule(dynamic) if(iNumThreads > 0)
```

So you can remove the OPENMVG_USE_OPENMP condition inside the loop